js/app.js

```diff
@@ -1 +1,188 @@
-console.log('Hello frontend');
+//Set all boxes to null
+//Create event lsitner that responds to a box being clicked
+//create event for each box
+//
+
+var playerTurn = 0;
+var setBoxValue = {
+	"a": null,
+	"b": null,
+	"c": null,
+	"d": null,
+	"e": null,
+	"f": null,
+	"g": null, 
+	"h": null,
+	"i": null
+}
+
+/*************************************************************************************************************************/
+
+//This function will take an input telling it what box was click populate the function and output an x or o
+	function fillBox(element){
+		document.getElementById(element).addEventListener("click", function(e) {
+   
+    		if(document.getElementById(element).textContent === ""){
+    			if(playerTurn % 2 === 0){
+    				document.getElementById(element).textContent = "X";
+    				setBoxValue[element] = "X";
+    				playerTurn++;
+    			}
+    			else if (playerTurn % 2 !== 0){
+    				document.getElementById(element).textContent= "O";
+                    setBoxValue[element] = "O";
+    				playerTurn++;
+    			} else {
+    				alert("This space is already taken");
+    			}
+                
+    		}
+    		Turn(playerTurn);
+            checkForWinner();
+    	});	
+        return setBoxValue;
+    }
+    fillBox("a");
+    fillBox("b");
+    fillBox("c");
+    fillBox("d");
+    fillBox("e");
+    fillBox("f");
+    fillBox("g");
+    fillBox("h");
+    fillBox("i");
+
+
+/***************************************************************************************************************************/
+    
+    function checkForWinner(){
+
+/*calculate if theres a win in the row [[a,b,c], [d,e,f], [g,h,i]];********************************************************/
+        if(setBoxValue["a"] === "X" && setBoxValue["b"] === "X" && setBoxValue["c"] === "X"){
```

Comment:
Make the "X" or "O" a variable and only checkForWinner for the player who's turn it is. Think about it this way - you wouldn't ever win a game on someone else's turn in tic-tac-toe. That will cut the amount of code in this function in half.